appengine/genetics/js/cage.js

```diff
@@ -0,0 +1,691 @@
+      break;
+    case 'acceptMate':
+      playerId = mouse.acceptMateOwner;
+      break;
+  }
+  var code = Genetics.Cage.players[playerId][1];
+  if (goog.isFunction(code)) {
+    code = code();
+  } else if (!goog.isString(code)) {
+    var player = Genetics.Cage.players[playerId][0];
+    throw 'Player ' + player + ' has invalid code: ' + code;
+  }
+  var interpreter = new Interpreter(code,
+      goog.partial(Genetics.Cage.initInterpreter, mouse, opt_suitor));
+  // Overwrite other function calls and call function we need return value of.
+  switch (mouseFunction) {
+    case 'pickFight':
+      interpreter.appendCode('proposeMate = null;');
+      interpreter.appendCode('acceptMate = null;');
+      interpreter.appendCode('pickFight();');
+      break;
+    case 'proposeMate':
+      interpreter.appendCode('pickFight = null;');
+      interpreter.appendCode('acceptMate = null;');
+      interpreter.appendCode('proposeMate();');
+      break;
+    case 'acceptMate':
+      interpreter.appendCode('pickFight = null;');
+      interpreter.appendCode('proposeMate = null;');
+      interpreter.appendCode('acceptMate(getSuitor());');
+      break;
+  }
+  return interpreter;
+};
+
+/**
+ * Inject the Genetics API into a JavaScript interpreter.
+ * @param {!Genetics.Mouse} mouse The mouse that is running the function.
+ * @param {!Genetics.Mouse|undefined} suitor The mouse passed as a parameter to
+ * the function (for acceptMate function call).
+ * @param {!Interpreter} interpreter The JS interpreter.
+ * @param {!Object} scope Global scope.
+ */
+Genetics.Cage.initInterpreter = function(mouse, suitor, interpreter,
+    scope) {
+  var pseudoMe = interpreter.UNDEFINED;
+  var pseudoSuitor = interpreter.ARRAY;
+  var pseudoAliveMice = interpreter.createObject(interpreter.ARRAY);
+  var aliveMice = Genetics.Cage.aliveMice_;
+  for (var i = 0; i < aliveMice.length; i++) {
+    var aliveMouse = aliveMice[i];
+    // Create a clone of alive mouse with string keys so that keys wont be
+    // renamed when compressed.
+    var clonedMouse = {
+      'pickFightOwner': aliveMice.pickFightOwner,
+      'proposeMateOwner': aliveMice.proposeMateOwner,
+      'acceptMateOwner': aliveMice.acceptMateOwner,
+      'sex': aliveMouse.sex,
+      'size': aliveMouse.size,
+      'startAggressiveness': aliveMouse.startAggressiveness,
+      'aggressiveness': aliveMouse.aggressiveness,
+      'startFertility': aliveMouse.startFertility,
+      'fertility': aliveMouse.fertility,
+      'age': aliveMouse.age,
+      'id': aliveMouse.id
+    };
+    var pseudoMouse = interpreter.nativeToPseudo(clonedMouse);
+    interpreter.setProperty(pseudoAliveMice, i, pseudoMouse);
+    // Check if the mouse running the interpreter has been created.
+    if (aliveMouse.id == mouse.id) {
+      pseudoMe = pseudoMouse;
+    }
+    // Check if the suitor parameter, if defined, has been created.
+    else if (suitor && aliveMouse.id == suitor.id) {
```

Comment:
Move comment inside if clause so that `} else if` can be on the same line.  Comment can then have 'Check if' removed since it is inside.
For consistency the previous comment can also be moved.